### PR TITLE
feat(sdk)!: rename `cloud.Counter` method `reset` to `set`, and add docs

### LINKED ISSUE #2654: `cloud.Counter` readme docs

### PR DESCRIPTION
**Pull request by Chriscbr**
Renames the method name from `reset` to `set` to better matches the mental model of assigning a customizable value to the counter. The numeric argument of `set` is also required now. Also adds some readme docs for cloud.Counter.

Closes #2654

BREAKING CHANGE: `cloud.Counter` method `reset` is now named `set`, and its argument is required (previously defaulted to the initial value)

## Checklist

- [ ] Title matches [Winglang's style guide](https://docs.winglang.io/contributors/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: docs/04-resources/counter.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+title: cloud.Counter 
+id: counter
+description: A built-in resource for representing an container for numbers in the cloud.
+keywords: [Wing reference, Wing language, language, Wing sdk, Wing programming language, Counter]
+---
+
+The `cloud.Counter` resource represents a stateful container for one or more numbers in the cloud.
```
Comment (hasanaburayyan) [inline docs/04-resources/counter.md:8]: Are these md files meant to document target state or current. I think they should be current, so maybe we dont say the `more` part just yet?

**Review by hasanaburayyan — APPROVED** [review]
Looks pretty "set" to me 😜

**Comment by ainvoner** [comment]
@skyrpex @polamoros this will also break the console. we need to track it.
@staycoolcall911 FYI

**Review by staycoolcall911 — APPROVED** [review]

File: docs/04-resources/counter.md (hunk 1 above)
Comment (Chriscbr) [reply docs/04-resources/counter.md:8]: Yes, good call.
```suggestion
The `cloud.Counter` resource represents a stateful container for a number in the cloud.
```

**Pull request merged**